crates/ide-assists/src/handlers/generate_getter.rs

```diff
@@ -72,88 +75,263 @@ pub(crate) fn generate_getter_mut(acc: &mut Assists, ctx: &AssistContext<'_>) ->
-    let field = ctx.find_node_at_offset::<ast::RecordField>()?;
+    // This if condition denotes two modes this assist can work in:
+    // - First is acting upon selection of record fields
+    // - Next is acting upon a single record field
+    //
+    // This is the only part where implementation diverges a bit,
+    // subsequent code is generic for both of these modes
+
+    let (strukt, info_of_record_fields, fn_names) = if !ctx.has_empty_selection() {
+        // Selection Mode
+        let node = ctx.covering_element();
+
+        let node = match node {
+            syntax::NodeOrToken::Node(n) => n,
+            syntax::NodeOrToken::Token(t) => t.parent()?,
+        };
+
+        let mut parent_struct: Option<ast::Struct> = None;
+        if let Some(struct_node) = node.ancestors().find(|it| ast::Struct::can_cast(it.kind())) {
+            parent_struct = ast::Struct::cast(struct_node);
+        }
+
+        // If we don't have a parent struct, something is wrong, OPERATION ABORT 🚨
+        if parent_struct.is_none() {
+            return None;
+        }
+
+        let (info_of_record_fields, field_names) =
+            extract_and_parse_record_fields(&node, ctx.selection_trimmed(), mutable)?;
+
+        (parent_struct.unwrap(), info_of_record_fields, field_names)
+    } else {
+        // Single Record Field mode
+        let strukt = ctx.find_node_at_offset::<ast::Struct>()?;
+        let field = ctx.find_node_at_offset::<ast::RecordField>()?;
 
-    let field_name = field.name()?;
-    let field_ty = field.ty()?;
+        let record_field_info = parse_record_field(field, mutable)?;
 
-    // Return early if we've found an existing fn
-    let mut fn_name = to_lower_snake_case(&field_name.to_string());
-    if mutable {
-        format_to!(fn_name, "_mut");
+        let fn_name = record_field_info.fn_name.clone();
+
+        (strukt, vec![record_field_info], vec![fn_name])
+    };
+
+    // No record fields to do work on :(
+    if info_of_record_fields.len() == 0 {
+        return None;
     }
-    let impl_def = find_struct_impl(ctx, &ast::Adt::Struct(strukt.clone()), fn_name.as_str())?;
+
+    // `find_struct_impl` looks for impl of a struct, but this also has additional feature
+    // where it takes a list of function names and check if they exist inside impl_, if
+    // even one match is found, it returns None
+    let impl_def = find_struct_impl(ctx, &ast::Adt::Struct(strukt.clone()), fn_names)?;
 
     let (id, label) = if mutable {
         ("generate_getter_mut", "Generate a mut getter method")
     } else {
         ("generate_getter", "Generate a getter method")
     };
-    let target = field.syntax().text_range();
+
+    let mut target: TextRange = info_of_record_fields[0].target;
+
+    // Computing collective text range of all record fields in selected region
+    info_of_record_fields
+        .clone()
+        .iter()
+        .for_each(|record_field_info| target = target.cover(record_field_info.target));
```

Comment:
```suggestion
    let target: TextRange = info_of_record_fields
        .iter()
        .map(|record_field_info| record_field_info.target)
        .reduce(|acc, target| acc.cover(target));
```

Reply:
Missed a `?` at the end, also nice, I can use a map to convert into new iterator and then reduce upon it, ah I really need to get a hang of using iterators properly 😅